### PR TITLE
deal with ipywidgets and javascript

### PR DESCRIPTION
**Pull request by basnijholt**

### CODE REVIEW

**Comment by jbweston** [comment]
@basnijholt did you put this here just to share it?

**Comment by basnijholt** [comment]
This will add the code to https://github.com/jupyter-widgets/jupyter-sphinx/pull/22.

But you can just copy the changes if you want.

**Comment by jbweston** [comment]
@basnijholt closing in favor of #2. I removed some bits e.g. the 'javascript' mimetype, which we don't strictly need now just for the ipywidgets support.

**Comment by jbweston** [comment]
also I made some minor modifications, e.g. only embedding widget data if there is any data

File: jupyter_sphinx/execute.py [hunk 1]
```diff
@@ -183,6 +185,13 @@ def cell_output_to_nodes(cell, data_priority, dir):
                     text=data,
                     format='html'
                 ))
+            elif mime_type.startswith('application'):
+                if mime_type.endswith('vnd.jupyter.widget-view+json'):
+                    data = json.dumps(data)
+                to_add.append(nodes.raw(
+                    text=f'<script type="{mime_type}">{data}</script>',
```
Comment (akhmerov) [inline jupyter_sphinx/execute.py]: Embedding in a script tag doesn't seem the right thing to do for most casts of `mime_type.startswith('application')`, see [this list](https://en.wikipedia.org/wiki/Media_type#Common_examples) for examples.

**Pull request closed without merging**